napari/layers/image/_image_constants.py

```diff
@@ -65,13 +65,18 @@ class Rendering(StringEnum):
             * iso: isosurface. Cast a ray until a certain threshold is
               encountered. At that location, lighning calculations are
               performed to give the visual appearance of a surface.
+            * iso_categorical: isosurface for categorical data (e.g., labels).
+              Cast a ray until a certain threshold is encountered. At that
+              location, lighning calculations are performed to give the visual
+              appearance of a surface.
             * average: average intensity projection. Cast a ray and display the
               average of values that were encountered.
     """
 
     TRANSLUCENT = auto()
     ADDITIVE = auto()
     ISO = auto()
+    ISO_CATEGORICAL = auto()
```

Comment:
Adding this here makes it available for standard volume rendering too... where it doesn't make sense (see image in main comment)

We could fix this by adding class methods for the subsets we want for each layer type (as we do for interpolation options in 2D) and displaying only those options on the Qt side
https://github.com/napari/napari/blob/78f6c87d88f0233e904f79190602cbe44806e5e7/napari/layers/image/_image_constants.py#L31-L39

Reply:
Wow, good catch. I didn't think to test that! That makes sense to me. Maybe a `view_image_subset` and `view_labels_subset`?

Reply:
Yep! although I'd go with `image_subset` and `labels_subset` in this case, the `view` in the example is to signify that all interpolation modes are available on the layer, but only this subset should be visible

Reply:
I've added a subset class method as suggested for both the Image and Labels layer. Then I populate the comboboxes with the relevant method.